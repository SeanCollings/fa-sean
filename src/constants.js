export const API_THE_ICONIC_URL =
  'https://eve.theiconic.com.au/catalog/products';

export const FILTER_SORT = {
  popularity: { display: 'Popularity', field: 'popularity' },
  new: { display: 'New Arrivals', field: 'new' },
  price_high: { display: 'Price High to Low', field: 'price_high' },
  price_low: { display: 'Price Low to High', field: 'price_low' },
  brand_asc: { display: 'Brand A to Z', field: 'brand_asc' },
  brand_desc: { display: 'Brand Z to A', field: 'brand_desc' },
};

export const FILTER_GENDER = {
  female: { display: 'Women', field: 'female' },
  male: { display: 'Men', field: 'male' },
  kids: { display: 'Kids', field: 'kids' },
};

export const COLOURS = {
  dark: '#232323',
  gray: '#eee',
  light: '#f7f7f7;',
  textDark: '#4d4d4d',
  textGray: '#909090',
  red: '#ef423c',
  white: '#ffffff',
  errorText: '#ee110d',
  errorBackground: '#fcecec',
};

// Â»
