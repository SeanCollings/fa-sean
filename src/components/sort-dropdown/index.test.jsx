import React from 'react';
import renderer from 'react-test-renderer';
import { useSelector } from 'react-redux';
import { useActions } from '../../hooks/use-actions';
import { FILTER_GENDER } from '../../constants';
import SortDropdown from '.';

jest.mock('react-redux');
jest.mock('../../hooks/use-actions');

describe('gender-fields unit test', () => {
  const mockUpdateFilter = jest.fn();
  useSelector.mockReturnValue({ gender: FILTER_GENDER.female });
  useActions.mockImplementation(() => ({
    updateFilter: mockUpdateFilter,
  }));

  test('should render', () => {
    const component = renderer.create(<SortDropdown />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
