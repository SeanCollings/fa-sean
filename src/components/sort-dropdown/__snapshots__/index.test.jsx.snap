// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gender-fields unit test should render 1`] = `
<div
  className="sc-dlfnbm ePhyqK"
>
  <div
    className="sc-hKgILt isxsAx"
  >
    <select
      className="sc-gsTCUz dmkZhF"
      id="sort"
      name="sort"
      onChange={[Function]}
    >
      <option
        value="popularity"
      >
        Popularity
      </option>
      <option
        value="new"
      >
        New Arrivals
      </option>
      <option
        value="price_high"
      >
        Price High to Low
      </option>
      <option
        value="price_low"
      >
        Price Low to High
      </option>
      <option
        value="brand_asc"
      >
        Brand A to Z
      </option>
      <option
        value="brand_desc"
      >
        Brand Z to A
      </option>
    </select>
  </div>
</div>
`;
